### PR TITLE
Add list options support for environments

### LINKED ISSUE #2257: Support pagination for ListEnvironments
Hey,

Would you consider supporting pagination for the ListEnvironments method? I appreciate that it is probably an edge case request but it would be nice to be able to pass in opts to the request for pagination in the instance where you have a lot of environments.

`func (s *RepositoriesService) ListEnvironments(ctx context.Context, owner, repo string, opts *ListOptions)`

### PR DESCRIPTION
**Pull request by chelnak**
Closes #2257 

This PR introduces an EnvironmentListOptions parameter for the ListEnvironments method.

Tests have also been updated to reflect the changes and an example has been added to the example directory.

### CODE REVIEW

File: example/listenvironments/main.go
Comment (gmlewis) [inline example/listenvironments/main.go:44]: This line is unnecessary because `log.Fatal` calls `os.Exit` for you.

File: example/listenvironments/main.go
Comment (gmlewis) [inline example/listenvironments/main.go:39]: same here... see below.

File: example/listenvironments/main.go
Comment (gmlewis) [inline example/listenvironments/main.go:31]:
```suggestion
	client := github.NewClient(tc)
```

And then lines 17-18 above can be deleted.

File: example/listenvironments/main.go
Comment (gmlewis) [inline example/listenvironments/main.go:19]: This line can be moved lower (around 35-36), closer to where it is actually being used.

File: example/listenvironments/main.go
Comment (gmlewis) [inline example/listenvironments/main.go:1]: Please add the standard header used in this repo, with a blank line, followed by a godoc comment like this:

```go
// listenvironments is an example of...
```

File: example/listenvironments/main.go
Comment (gmlewis) [inline example/listenvironments/main.go:15]: Please move these lines immediately before the `package main` above (but separated from the initial comment required in this repo):

```go
// Copyright 2022 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

// listenvironments is an example of how to use the ListEnvironments method with EnvironmentListOptions.
// It's runnable with the following command:
//     export GITHUB_TOKEN=your_token
//     export GITHUB_REPOSITORY_OWNER=your_owner
//     export GITHUB_REPOSITORY_NAME=your_repo
//     go run .
package main
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @chelnak !
Just a few changes are needed for the example... then we will be ready for a second LGTM before merging.

File: example/listenvironments/main.go
Comment (chelnak) [reply example/listenvironments/main.go]: Thank you - i've removed this!

File: example/listenvironments/main.go
Comment (chelnak) [reply example/listenvironments/main.go]: Thank you - i've removed this!

File: example/listenvironments/main.go
Comment (chelnak) [reply example/listenvironments/main.go]: Updated and removed the var declaration

File: example/listenvironments/main.go
Comment (chelnak) [reply example/listenvironments/main.go]: Moved as suggested. I've also swapped the var declaration for the short declaration here too.

File: example/listenvironments/main.go
Comment (chelnak) [reply example/listenvironments/main.go:12]: Done ✅

File: example/listenvironments/main.go
Comment (chelnak) [reply example/listenvironments/main.go]: Done ✅

**Comment by chelnak** [comment]
@gmlewis All of your comments should now be resolved. Thanks for taking the time to review.

File: example/listenvironments/main.go
Comment (gmlewis) [inline example/listenvironments/main.go:1]:
```suggestion
// Copyright 2022 The go-github AUTHORS. All rights reserved.
```

The copyright date is set to the year that the file first appears in our repo, so please change this to 2022.  Thanks.

Comment (chelnak) [reply example/listenvironments/main.go]: My bad - committed the suggestion.

**Review by gmlewis — APPROVED** [review]
Thank you, @chelnak !
LGTM.
Awaiting second LGTM before merging.

**Comment by chelnak** [comment]
Hello! Is there anything else you would like me to change for this PR?

**Comment by gmlewis** [comment]
> Hello! Is there anything else you would like me to change for this PR?

No, thank you.  We are awaiting any other contributor to this repo to give an LGTM and approve this PR before merging.

**Comment by chelnak** [comment]
Hey hey! Hope everything is going well! Wondering if we could request a review from another contributor 🙏? Appreciate everyone is super busy though.

**Comment by gmlewis** [comment]
Sorry for the delay, @chelnak !
Hopefully by tomorrow we will be able to get this one merged. Maybe @Parker77 has time to review?

**Comment by chelnak** [comment]
Thank you 🙏 😄

**Review by Parker77 — APPROVED** [review]
LGTM.

**Comment by gmlewis** [comment]
Than you, @Parker77 !
Merging.

**Comment by fun4cash** [comment]
import "github.com/google/go-github/v43/github"

**Pull request merged**
